src/transformers/generation/logits_process.py

```diff
@@ -657,7 +657,7 @@ def _set_scores_to_inf_for_banned_tokens(
                 # [ 1  0  0 ]
 
                 banned_mask = (
-                    torch.sparse.LongTensor(banned_mask.t(), indices, scores.size())
+                    torch.sparse.LongTensor(banned_mask.to(indices.device).t(), indices, scores.size())
```

Comment:
It would be more efficient to create `banned_mask` on the proper device. Can you put this in the creation of `banned_mask` (and switch the order in which indices and banned_mask are created)?

Reply:
ok